etl/steps/data/garden/health/2023-04-25/wgm_2018.py

```diff
@@ -0,0 +1,353 @@
+from owid.catalog import Dataset, Table
+from shared import MAPPING_COLUMN_NAMES, MAPPING_GENDER_VALUES, MAPPING_QUESTION_VALUES
+from structlog import get_logger
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+# Minimum number of participants answering a question. Questions for demographic groups with fewer answers are filtered out.
+# This mostly affects very granular breakdowns (e.g. by age and gender)
+THRESHOLD_ANSWERS = 100
+
+
+def run(dest_dir: str) -> None:
+    log.info("wgm_2018: start")
+
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow: Dataset = paths.load_dependency("wgm_2018")
+
+    # Read table from meadow dataset.
+    tb_meadow = ds_meadow["wgm_2018"]
+
+    # Create a dataframe with data from the table.
+    df = pd.DataFrame(tb_meadow)
+
+    #
+    # Process data.
+    #
+    log.info("wgm_2018: harmonize_countries")
+    # Add age_group (based on age value)
+    df = add_age_group(df)
+    # Keep relevant columns, unpivot dataframe, harmonize country names.
+    df = clean_df(df)
+    # Merge "Don't Know" and "Refused" categories, as done in WGM 2020
+    df = merge_dk_and_refused_answers(df)
+    # Build dataframe with shares of answers to each questions.
+    df = make_df_with_share_answers(df)
+    # Map IDs to labels (question, age and gender groups, answers)
+    df = map_ids_to_labels(df)
+    # Filter out questions with low participation
+    df = filter_rows_with_low_participation(df)
+    # Format dataframe with appropriate columns, indexes
+    df = final_formatting(df)
+    # Create a new table based on the dataframe `df`
+    tb_garden = Table(df, short_name=paths.short_name, underscore=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(dest_dir, tables=[tb_garden], default_metadata=ds_meadow.metadata)
+    ds_garden.update_metadata(paths.metadata_path)
+    # Add explanation to dataset description
+    ds_garden.metadata.description += (
+        "\n\nNote: Data for answers that where the demographic group had less than 100 participants are filtered out."
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+    log.info("wgm_2018: end")
+
+
+def add_age_group(df: pd.DataFrame) -> pd.DataFrame:
+    """Add age_group column.
+
+    Age groups are defined as follows:
+    - 15-29
+    - 30-49
+    - 50-64
+    - 65+
+    - DK/Refused
+    """
+    log.info("wgm_2018: add_age_group")
+    df["age_group"] = df["age"].apply(_map_age_to_group)
+    return df
+
+
+def _map_age_to_group(age_str: str):
+    if age_str == " ":
+        return "DK/Refused"
+    age = int(age_str)
+    if age < 30:
+        return "15-29"
+    if age < 50:
+        return "30-49"
+    if age < 65:
+        return "50-64"
```

Comment:
This function would be a bit safer if all cases were considered, using if, elif, and a final else to raise an error. There may be some negative ages by mistake, and they would be assigned "+65".